src/test/compile-fail/for-loop-refutable-pattern-error-message.rs

```diff
@@ -9,7 +9,5 @@
 // except according to those terms.
 
 fn main() {
-    for
-        &1 //~ ERROR refutable pattern in `for` loop binding
-        in [1].iter() {}
+    for &1 in [1].iter() {} //~ ERROR refutable pattern in `for` loop binding
```

Comment:
It is in some sense a regression, but it is due to a more accurate span translation - the actual match arm that gets produced has the pattern `Some(&1)`. The generated `&1` gets the same span as the original, the `Some(...)` used to get that span too. Now it doesn't - only code which is preserved gets the preserved spans, the closest span is for the whole expression, which is what gives the less precise error message. I believe this setup is more accurate and better for tooling, although marginally worse for error messages.